src/pm-conversations/__tests__/pmConversationsSelectors-test.js

```diff
@@ -29,45 +29,33 @@ describe('getRecentConversations', () => {
   });
 
   test('returns unique list of recipients, includes conversations with self', () => {
-    const meAndJohnPm1 = eg.pmMessage({ id: 1, recipients: [eg.selfUser, userJohn] });
-    const meAndMarkPm = eg.pmMessage({ id: 2, recipients: [eg.selfUser, userMark] });
-    const meAndJohnPm2 = eg.pmMessage({ id: 3, recipients: [eg.selfUser, userJohn] });
-    const meOnlyPm = eg.pmMessage({ id: 4, recipients: [eg.selfUser] });
-    const meJohnAndMarkPm = eg.pmMessage({ id: 0, recipients: [eg.selfUser, userJohn, userMark] });
-
     const state = eg.reduxState({
       realm: eg.realmState({ email: eg.selfUser.email }),
       users: [eg.selfUser, userJohn, userMark],
       narrows: Immutable.Map({
-        [ALL_PRIVATE_NARROW_STR]: [
-          meJohnAndMarkPm.id,
-          meAndJohnPm1.id,
-          meAndMarkPm.id,
-          meAndJohnPm2.id,
-          meOnlyPm.id,
-        ],
+        [ALL_PRIVATE_NARROW_STR]: [0, 1, 2, 3, 4],
```

Comment:
There is a tradeoff here, but I think on net this version is a bit easier to understand.

One key condition of that is that these IDs all stay local -- their meaning is all contained within the test case, and doesn't involve looking out to anything further afield. (I'd like to say the scope of code that can care about one of these IDs all fits on a screen, but that may not be true on laptop screens. It'd be better if each test's code were shorter, so that it did.)

Then on the other hand these names were feeling like a bit of work to decode in themselves. Partly because they're long, perhaps? They're sensible names, and I don't have better ideas for names for these values, so that seems potentially inherent to naming them in these tests.

... In fact. It looks like the data don't actually match up with each other!
```js
          {
            sender_id: userJohn.user_id,
            unread_message_ids: [meAndMarkPm1.id, meAndMarkPm2.id],
          },
          {
            sender_id: userMark.user_id,
            unread_message_ids: [meAndJohnPm1.id],
          },
```
The fact that we didn't notice that until now suggests to me that the names weren't providing a lot of value in making this test data easy to understand. I'll add a fix for that to this branch.

Another thing on the other hand is that the ordering of the message IDs is important -- the output is supposed to be sorted by message ID (descending), and the second test is all about that property. So with the names you have to go consulting the definitions and matching them up to see that the ordering is right, whereas with the IDs it's immediate.

Reply:
Ah OK, makes sense. And thanks for catching that mismatch!